### ISSUE TITLE
Clone over ssh with ssh key from file fails in a docker image

### ISSUE THREAD

**Issue by albertsun** [libgit2/rugged#530]
Wonder if anyone might know what's happening here with this command or be able to help me debug. I don't know my way around libgit2 or libssh2 internals that well yet.

I have a Rails app that I'd like to run in a Docker image. From within that when I run this command it fails.

```
[4] pry(main)> credentials = Rugged::Credentials::SshKey.new({
[4] pry(main)* :username => "git", # TODO: this could be picked up from connection string
[4] pry(main)* :publickey => "/usr/src/app/config/keys/testcapitalgit.pub",
[4] pry(main)* :privatekey => "/usr/src/app/config/keys/testcapitalgit",
[4] pry(main)* :passphrase => "capital_git passphrase"
[4] pry(main)* })
=> #<Rugged::Credentials::SshKey:0x007f5354f72778
 @passphrase="capital_git passphrase",
 @privatekey="/usr/src/app/config/keys/testcapitalgit",
 @publickey="/usr/src/app/config/keys/testcapitalgit.pub",
 @username="git">
[5] pry(main)>
[6] pry(main)> opts = {
[6] pry(main)* credentials: credentials
[6] pry(main)* }
=> {:credentials=>
  #<Rugged::Credentials::SshKey:0x007f5354f72778
   @passphrase="capital_git passphrase",
   @privatekey="/usr/src/app/config/keys/testcapitalgit",
   @publickey="/usr/src/app/config/keys/testcapitalgit.pub",
   @username="git">}
[7] pry(main)> repo = Rugged::Repository.clone_at("git@github.com:newsdev/capital_git_testrepo.git", local_path, opts)
Rugged::SshError: Failed to authenticate SSH session: Callback returned error
```

That same command with the same ssh keys works successfully from my OS X machine and from a separate Ubuntu machine.

The Dockerfile, testcase and output from ldd are here: https://gist.github.com/albertsun/220d86d29dc38ce9668f

Any help or guidance in where to look to try and debug is much appreciated!

Albert

**Comment by buth** [comment]
I can repeat this issue compiling against libssh2-1.4.x, but it seems to be fixed with libssh2-1.6.x. The former is the current in the Debian stable repo; the latter is the current in Homebrew's repo.

The underlying problem seems to be related to decrypting the private key. I was able to get the same error using an incorrect password.

**Comment by albertsun** [comment]
Ok confirmed that using libssh2-1.6.0 compiled from source works. Closing this.
